apps/zotonic_core/src/db/z_db.erl

```diff
@@ -749,20 +827,32 @@ update_map_atom_arrays(Props) ->
         end,
         Props).
 
-update_merge_props(DbDriver, C, QTab, Id, NewProps) when is_map(NewProps) ->
-    QTab1 = z_convert:to_list(QTab),
-    Merged = case equery1(DbDriver, C, "select props from "++QTab1++" where id = $1", [Id]) of
-        {ok, OldProps} when is_list(OldProps) ->
-            maps:merge(z_props:from_props(OldProps), NewProps);
-        {ok, OldProps} when is_map(OldProps) ->
-            maps:merge(OldProps, NewProps);
+update_merge_props(DbDriver, Connection, Table, Cols, Id, #{ <<"props_json">> := NewProps }=UpdateProps, Context) ->
+    P = case get_current_props(DbDriver, Connection, Table, Id, Context) of
+            {ok, OldProps} ->
+                UpdateProps#{<<"props_json">> => ?DB_PROPS_JSON( maps:merge(OldProps, NewProps) ) };
+            _ ->
+                UpdateProps#{<<"props_json">> => ?DB_PROPS_JSON( NewProps )}
+        end,
+
+    %% Clear the existing props column
+    case lists:member(<<"props">>, Cols) of
+        true -> P#{ <<"props">> => null };
```

Comment:
Shouldn't that be `undefined` ?

Reply:
I thought that the old props will be removed when it is set to null.